src/autopas/utils/AutoPasConfigurationCommunicator.h

```diff
@@ -75,6 +78,53 @@ void distributeConfigurations(std::set<ContainerOption> &containerOptions, Numbe
                               std::set<DataLayoutOption> &dataLayoutOptions, std::set<Newton3Option> &newton3Options,
                               int rank, int commSize);
 
+/**
+ * Distribute ranks in buckets, which contain only ranks with similar scenarios.
+ * Each bucket then has its own search space.
+ * @param comm
+ * @param rank
+ * @param commSize
+ * @param bucket
+ */
+template<class Particle>
+void distributeRanksInBuckets(AutoPas_MPI_Comm comm, AutoPas_MPI_Comm *bucket, const std::shared_ptr<autopas::ParticleContainerInterface<Particle>>& container) {
+  int rank;
+  AutoPas_MPI_Comm_rank(comm, &rank);
+  int commSize;
+  AutoPas_MPI_Comm_size(comm, &commSize);
+
+  auto *homogeneities_pointer = static_cast<double *>(malloc(commSize * sizeof(double)));
```

Comment:
C++:
Whenever you call `malloc`, there HAS to be a matching `free` call!
Same for `new`, which requires a `delete`!
Otherwise, you are leaking memory, because you allocate sth. on the heap and never free it again.

As this is cumbersome to track, please do NOT use `malloc` or `new` within autopas.
Instead, use [RAII](https://en.wikipedia.org/wiki/Resource_acquisition_is_initialization), i.e., there should be an object that manages the lifetime of the allocated heap memory.
Here, you can simply use a `std::vector`.
```suggestion
  std::vector<double> homogeneities(commSize);
```

Reply:
solved with commit "dont use malloc"